source/includes/connect/atlas-connection.java

```diff
@@ -0,0 +1,44 @@
+import com.mongodb.*;
+import com.mongodb.reactivestreams.client.MongoCollection;
+
+import org.bson.BsonDocument;
+import org.bson.BsonInt64;
+import org.bson.Document;
+
+import com.mongodb.reactivestreams.client.MongoClient;
+import com.mongodb.reactivestreams.client.MongoClients;
+import com.mongodb.reactivestreams.client.MongoDatabase;
+import org.bson.conversions.Bson;
+import org.reactivestreams.Publisher;
+
+public class updateDocument {
+    public static void main(String[] args) {
+        // Replace the placeholder with your Atlas connection string
+        String uri = "<connection string>";
+
+        // Construct a ServerApi instance using the ServerApi.builder() method
+        ServerApi serverApi = ServerApi.builder()
+                .version(ServerApiVersion.V1)
+                .build();
+
+        MongoClientSettings settings = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(uri))
+                .serverApi(serverApi)
+                .build();
+
+        // Create a new client and connect to the server
+        try (MongoClient mongoClient = MongoClients.create(settings))
+        {
+            MongoDatabase database = mongoClient.getDatabase("<database name>");
+            MongoCollection<Document> collection = database.getCollection("<collection name>");
+
+            try {
+                Bson command = new BsonDocument("ping", new BsonInt64(1));
+                Publisher<Document> commandResult = database.runCommand(command);
```

Comment:
So this is where Reactive Streams get tricky. This is because they don't do anything unless subscribed to and data is requested. Explaining this to users, is also cumbersome as you have to include things like special subscribers that block until a result is collected. etc.

`Publishers` also have a way of passing errors via the `Subscriber`, so generally try catches are used. `Subscribers` have the following signature:

```
onSubscribe(Subscription s) - what to do when subscribed to - eg request a result
onNext(T t) - what to do with each published item
onError(Throwable t) - What to do if there is an error
onComplete() - What to do on complete.
```

So how to make this a useful example to end users without also having to write lots of supporting code that would also need explaining?  The best answer I can come up with is use a framework that is designed to handle Reactive Streams.

As Reactive Streams is a low level library, most (if not all) consumers of reactive streams, will use a library to handle them. This both improves the ergonomics of reactive streams, the accessiblity of them and correctness of use.  For example internally we use [Project Reactor](https://projectreactor.io/docs/core/release/reference/).

I recommend that all examples use this library, so that the examples are clear and any abstractions like subscribing are handled by reactor. So the example becomes:

```
    Mono.from(database.runCommand(command))
        .doOnSuccess(x ->  System.out.println("Pinged your deployment. You successfully connected to MongoDB!"))
        .doOnError(err -> System.out.println("Error: " + err.getMessage()))
        .block();
```

Reply:
Hi Ross,

I believe I implemented what you were talking about with my new pushes, but let me know if you were thinking of something else! Also, I had a clarifying question. In my other PRs, I've been using a specific subscriber implementation, the [SubscriberHelpers](https://github.com/mongodb/mongo-java-driver/blob/master/driver-reactive-streams/src/examples/reactivestreams/helpers/SubscriberHelpers.java), which I believe were written by someone in the Java team a while ago. To confirm, you believe that using this implementation/asking users to implement it is not as ideal, due to asking users to write lots of supporting code/explaining it, and you recommend that we use the outside Project Reactor library instead to implement reactive streams?